### ISSUE TITLE
New Check UnusedTryResourceShouldBeUnnamed

### ISSUE THREAD

**Issue by mahfouz72** [checkstyle/checkstyle#15484]
child of #14942
Violate the non-use of try resources. They should be unnamed.

Originally proposed to extend `UnusedLocalVariableCheck`  in [#15024](https://github.com/checkstyle/checkstyle/issues/15024) but looks like the majority is voting for a new check to achieve this behaviour.

---

**Config**

```XML
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="UnusedTryResourceShouldBeUnnamed"/>
    </module>
</module>
```

---

**Example**

```Java
public class Test {
     void test() {
         // violation below, 'Unused try resource 'a' should be unnamed'
         try (var a = lock()){
             
         } catch (Exception e) {
             
         }
         
         // ok below, declared as unnamed
         try (var _ = lock()){
             
         } catch (Exception e) {
             
         }
     }
     
     AutoCloseable lock() {
         return null;
     }
}
```

---

**Expected Output**

```
$ java -jar checkstyle-10.18.0-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] D:\Test.java:4:15:Unused try resource should be unnamed [UnusedTryResourceShouldBeUnnamed]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
do we expect such check to be very different from UnusedLambdaParameterShouldBeUnnamed ?

**Comment by mahfouz72** [comment]
No, almost the same

**Comment by romani** [comment]
@nrmancuso, is it reasonable to combine them in one Check and manage target by "tokens" ?
There prons and cons ..... for both approaches.

**Comment by rnveach** [comment]
> is it reasonable to combine them

Not by their current check name. A lambda is not a try-with-resource, and vice versa.

**Comment by rnveach** [comment]
I am good with first post.

**Comment by nrmancuso** [comment]
> @nrmancuso, is it reasonable to combine them in one Check and manage target by "tokens" ? There prons and cons ..... for both approaches.

It is better to not create another JavadocMethod; small, simple, easy to reason about checks make for easy maintenance. Checking for some unused variable will have different nuances for each type (lambda, catch parameter, etc).

**Comment by romani** [comment]
ok, lets do separate.
code sharing is technical issue and can be resolved easily, separate testing and full independent coverage will help  us to not loose us functional coverage of them individually.

**Comment by akankshat05** [comment]
@romani Can I work on this?

**Comment by mahfouz72** [comment]
@akankshat05  Yes. just send a PR. The issue is still open and no one is working on it

**Comment by romani** [comment]
Everyone is free to choose any approved issue.
@akankshat05, I see your contributions in our project, and we all appreciate this.
I just noticed that you didn't not try any issue that  is labeled as "good forth issue". I highly recommend sending PR for it before starting new Check.
I just try to let you have smooth experience with growing complexity. Knowledge that you gain in such issues will help you a lot.

**Comment by romani** [comment]
@kkoutsilis, please help us to make this Check.

**Comment by kkoutsilis** [comment]
@romani I have limited availability for the next month or so, but I will work on it when I have time. Thank you

**Comment by romani** [comment]
No problem, 😁 pretty sure issue can wait few months, no pressure

**Comment by romani** [comment]
Check is merged
